orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/metrics/AtlasQueueMonitor.kt

```diff
@@ -49,8 +50,12 @@ class AtlasQueueMonitor
   @EventListener
   fun onQueueEvent(event: QueueEvent) {
     when (event) {
-      is QueuePolled -> _lastQueuePoll.set(clock.instant())
-      is MessageProcessing -> _messageLags.add(event.lag)
+      QueuePolled -> _lastQueuePoll.set(clock.instant())
+      is MessageProcessing -> {
+        _meanMessageLag.updateAndGet { it + event.lag.toMillis() }
```

Comment:
I think updateAndGet here has a concurrency issue if `it + event.lag.toMillis()` isn't side-effect free and is operating on the underlying list object. Could be I'm wrong if the kotlin `+` is returning a new list object with the result appended..